src/data/turbo-redis-data-source.ts

```diff
@@ -0,0 +1,409 @@
+    region?: Region;
+  }): Promise<ContiguousData> {
+    // TODO: Configuration for whether metadata or offsets are checked first
+    try {
+      const offsetsInfo = await this.getCachedTurboOffsetsInfo(id);
+      if (offsetsInfo) {
+        this.log.debug(`Turbo Elasticache: Found offsets for ${id}`, {
+          offsetsInfo,
+        });
+
+        const {
+          parentDataItemId,
+          parentPayloadDataStart,
+          startOffsetInRawParent,
+          rawContentLength,
+          payloadContentType,
+          payloadDataStart,
+        } = offsetsInfo;
+
+        const startOffsetInParentPayload =
+          startOffsetInRawParent + payloadDataStart - parentPayloadDataStart;
+        const payloadLength = rawContentLength - payloadDataStart;
+
+        const nestedDataItemDataStream = await this.getData({
+          id: parentDataItemId,
+          region: {
+            offset: startOffsetInParentPayload + (region?.offset ?? 0),
+            size: region?.size ?? payloadLength,
+          },
+        });
+        if (nestedDataItemDataStream?.stream === undefined) {
+          const errMsg = `Turbo Elasticache: Parent ${parentDataItemId} payload data not found for nested data item ${id}`;
+          this.log.debug(errMsg, {
+            offsetsInfo,
+          });
+          throw new Error(errMsg);
+        }
+
+        this.log.debug(
+          `Turbo Elasticache: Found parent ${parentDataItemId} payload data stream for ${id}. Returning nested data item from offsets info.`,
+          {
+            offsetsInfo,
+          },
+        );
+
+        // track &&
+        //   turboPayloadFetchResultTotal.inc({
+        //     result: 'hit',
+        //     store: 'cache',
+        //     form: 'offsets',
+        //   });
+
+        const requestAttributesHeaders =
+          generateRequestAttributes(requestAttributes);
+
+        return {
+          stream: nestedDataItemDataStream.stream,
+          size: region?.size ?? payloadLength,
+          sourceContentType: payloadContentType,
+          verified: false,
+          trusted: true,
+          cached: false,
+          requestAttributes: requestAttributesHeaders?.attributes,
+        };
+      }
+
+      // track &&
+      //   turboPayloadFetchResultTotal.inc({
+      //     result: 'miss',
+      //     store: 'cache',
+      //     form: 'offsets',
+      //   });
+
+      const metadata = await this.getCachedTurboMetadata(id);
+      if (metadata) {
+        this.log.debug(`Turbo Elasticache: Found metadata for ${id}`, {
+          metadata,
+        });
+        return this.getCachedTurboPayloadDataStreamFromMetadata({
+          dataItemId: id,
+          payloadContentType: metadata.payloadContentType,
+          payloadStartOffset: metadata.payloadStartOffset,
+          region,
+          requestAttributes,
+        });
+        // .then((dataStream) => {
+        //   TODO: Distinguish between 'cache' and 'fs' hits when fs is introduced
```

Comment:
`ReadThroughDataCache` will already do some caching of any data source it wraps. Is the future FS cache referred to here something beyond that?